easybuild/easyblocks/b/boost.py

```diff
@@ -120,10 +125,35 @@ def configure_step(self):
             # configure the boost mpi module
             # http://www.boost.org/doc/libs/1_47_0/doc/html/mpi/getting_started.html
             # let Boost.Build know to look here for the config file
-            f = open('user-config.jam', 'a')
-            f.write("using mpi : %s ;" % os.getenv("MPICXX"))
-            f.close()
 
+            txt = ''
+            # Check if using a Cray toolchain and configure MPI accordingly
+            if self.toolchain.name.startswith('Cray'):
+                if self.toolchain.PRGENV_MODULE_NAME_SUFFIX is 'gnu':
+                    craympichdir = os.getenv('CRAY_MPICH2_DIR')
+                    craygccversion = os.getenv('GCC_VERSION')
+                    txt = '\n'.join([    
+                        'local CRAY_MPICH2_DIR =  %s ;' % craympichdir,
+                        'using gcc ',
+                        ': %s' % craygccversion,
+                        ': CC ',
+                        ': <compileflags>-I$(CRAY_MPICH2_DIR)/include ',
+                        '  <linkflags>-L$(CRAY_MPICH2_DIR)/lib \ ',
+                        '; ',
+                        'using mpi ',
+                        ': CC ',
+                        ': <find-shared-library>mpich ',
+                        ': %s' % self.cfg['mpi_launcher'],
+                        ';',
+                        '',
+                    ])
+                else: 
+                    raise EasyBuildError("Bailing out: PrgEnv-intel and PrgEnv-cray not supported.")
+            else:
+                txt = "using mpi : %s ;" % os.getenv("MPICXX")
+
+        write_file('user-config.jam', txt, append=True)
```

Comment:
@gppezzi indentation seems off here? should be intended one level deeper, so it's in the `if self.cfg['boost_mpi']:`, otherwise `txt` is undefined